sqlglot/dialects/bigquery.py

```diff
@@ -357,17 +357,29 @@ class Parser(parser.Parser):
             "OPTIONS": lambda self: exp.Properties(expressions=self._parse_with_property()),
         }
 
-        def _parse_table_part(self, schema: bool = False) -> t.Optional[exp.Expression]:
-            this = super()._parse_table_part(schema=schema)
+        def _parse_table_part(
+            self, index: int = 0, schema: bool = False
+        ) -> t.Optional[exp.Expression]:
+            this = super()._parse_table_part(index=index, schema=schema)
 
             # https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#table_names
-            if isinstance(this, exp.Identifier):
+            # table name that is an unquoted identifier can additionally include single dashes
+            # only the first identifier in the table path can have dashes
+            if index == 0 and isinstance(this, exp.Identifier) and not this.quoted:
```

Comment:
i don't know if we need index checks

Reply:
I guess we can go without, but it is more correct and actually helps to isolate the logic. The first and second are orthogonal.